src/analysis/normed_space/exponential.lean

```diff
@@ -431,11 +438,48 @@ begin
   { rw [succ_nsmul, pow_succ, exp_add_of_commute ((commute.refl x).smul_right n), ih] }
 end
 
+variables (𝕂)
+
+/-- Any continuous ring homomorphism commutes with `exp`. -/
+lemma map_exp {F} [ring_hom_class F 𝔸 𝔹] (f : F) (hf : continuous f) (x : 𝔸)  :
+  f (exp 𝕂 𝔸 x) = exp 𝕂 𝔹 (f x) :=
+map_exp_of_mem_ball f hf x $ (exp_series_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _
+
+lemma exp_smul {G} [monoid G] [mul_semiring_action G 𝔸] [has_continuous_const_smul G 𝔸]
+  (g : G) (x : 𝔸) :
+  exp 𝕂 𝔸 (g • x) = g • exp 𝕂 𝔸 x :=
+(map_exp 𝕂 (mul_semiring_action.to_ring_hom G 𝔸 g) (continuous_const_smul _) x).symm
+
+lemma exp_units_conj (y : 𝔸ˣ) (x : 𝔸)  :
+  exp 𝕂 𝔸 (y * x * ↑(y⁻¹)) = y * exp 𝕂 𝔸 x * ↑(y⁻¹) :=
+exp_smul _ (conj_act.to_conj_act y) x
+
+lemma exp_units_conj' (y : 𝔸ˣ) (x : 𝔸)  :
+  exp 𝕂 𝔸 (↑(y⁻¹) * x * y) = ↑(y⁻¹) * exp 𝕂 𝔸 x * y :=
+exp_units_conj _ _ _
+
+@[simp] lemma prod.fst_exp [complete_space 𝔹] (x : 𝔸 × 𝔹) : (exp 𝕂 (𝔸 × 𝔹) x).fst = exp 𝕂 𝔸 x.fst :=
+map_exp _ (ring_hom.fst 𝔸 𝔹) continuous_fst x
+
+@[simp] lemma prod.snd_exp [complete_space 𝔹] (x : 𝔸 × 𝔹) : (exp 𝕂 (𝔸 × 𝔹) x).snd = exp 𝕂 𝔹 x.snd :=
+map_exp _ (ring_hom.snd 𝔸 𝔹) continuous_snd x
+
+/- At the time of writing _statement_ times out, but might not in future:
+```lean
+@[simp] lemma pi.exp_apply {ι : Type*} {𝔸 : ι → Type*} [fintype ι] [Π i, normed_ring (𝔸 i)]
+  [Π i, normed_algebra 𝕂 (𝔸 i)]
+  [Π i, complete_space (𝔸 i)] (x : Π i, 𝔸 i) (i : ι) :
+  (exp 𝕂 (Π i, 𝔸 i) x : Π i, 𝔸 i) i = exp 𝕂 (𝔸 i) (x i) :=
+sorry
+```
+-/
```

Comment:
#13444 will likely change the elaboration here dramatically anyway - I figured it was worth leaving this in as comment, so that I can try uncommenting it in a future PR without having to work out the typeclass requierments again.